### ISSUE TITLE
Please post code on pypi for usage with pip.

### ISSUE THREAD

**Issue by thorsummoner** [jleclanche/python-xdg#1]
I didn't see this up on https://pypi.python.org/pypi, and `python -m pip install --user python-xdg` failed as:

```
Downloading/unpacking python-xdg
  Could not find any downloads that satisfy the requirement python-xdg
Cleaning up...
No distributions at all found for python-xdg
Storing debug log for failure in /tmp/tmpKJCHBJ
```

I would prefer to install your package via pip, so I'm requesting that.

**Comment by jleclanche** [comment]
The package isn't maintained enough to warrant going on pypi. It's very feature-incomplete and I am not focusing on it currently.

**Comment by pjz** [comment]
I'll second this; I can't find any other library that will let me do:

    from xdg.mime import MimeType
    icon_type = MimeType.fromName(filename).icon()

Also: if you package it and publish it, you'll draw more contributors who might help maintain it.
AlsoAlso:  if it's feature incomplete now, open some issues on it about what features you think it might need to be releasable; maybe someone will give you a PR.

**Comment by thorsummoner** [comment]
I'd be happy to maintain it honestly, I feel this library is an essential
component of desktop python computing

**Comment by jleclanche** [comment]
Have you guys looked at https://freedesktop.org/wiki/Software/pyxdg/ ? I know it's an old lib with a legacy API but it got a new, very open maintainer (@takluyver) recently. It's the "official" xdg library, so I'd rather draw contributors there. When I build this one, pyxdg was completely unmaintained and barely compatible with Python 2.7, let alone 3.x. This has changed.

**Comment by jleclanche** [comment]
I added a license so you guys can do whatever you want with the code. I also added contributing guidelines/code style, I'll merge good PRs that comply to those... but don't expect me to work on this much myself.

**Comment by takluyver** [comment]
Maintenance help with PyXDG is also welcome - it hasn't had an update in a few years, and I'm not currently using it for any other projects myself, so I don't have much reason to improve it. However, being an old library, I am very keen not to break the existing APIs, so I'll be fairly conservative about what goes in.

If people are more interested in new APIs, I think it might make more sense to have different Python packages for different XDG specs - e.g. an `xdg_mime` or `xdg_icon_theme` package rather than an overall xdg package. XDG is an umbrella group that maintains a number of distinct specifications, and there are bits of PyXDG that I'm pretty sure are essentially unused (like 'recent files').

**Comment by jleclanche** [comment]
@takluyver a common pain and want is to be able to just do "What's the mime type for this file? (no really don't just look at its extension" and "What's the program to open that mime type?" (and furthermore, "How do I launch that program?". A `requests`-like package to do just that would be awesome.

This package here could do that. I'm ok breaking backwards compatibility on it since there's very few users ... and it's using some of my old, less-consistent code style so I'll be renaming functions to under_score style anyway if we go forward.

**Comment by takluyver** [comment]
> What's the mime type for this file? (no really don't just look at its extension)

In PyXDG, this is `xdg.Mime.get_type_by_contents()`

**Comment by jleclanche** [comment]
Yeah; what i meant was you don't want to use python's dumb `mimetype` module.
